### PR TITLE
Fix keyboard shortcut bug 

### LINKED ISSUE #2919: Pressing enter / shift-enter on a subworksheets block item gives an error
If I use j/k to scroll down to a subworksheets block and then press enter / shift-enter, I get:

image

image

This is probably because `.refs` is no longer defined -- we need to probably replace this by getting elements by id.

### LINKED ISSUE #2939: hitting 'enter' on a list of worksheets on a worksheets does nothing
This should go to the focused worksheet.
Screenshot from 2020-10-06 12-25-31

### LINKED ISSUE #2951: Keyboard Shortcut Issue
- [x] Sometimes ‘k’ skips over a bundle, though ‘up’ works; sometimes happens after pasting in a bundle

- [x] Editing schema - ‘enter’ shouldn’t add a space
updated: 
The reason why this happens is we manually replace `space`  with `_` when editing the schema name since `space`  is not allowed in a legal schema name. [code](https://github.com/codalab/codalab-worksheets/blob/c9197db059a7c2146be643ad9fc293bfd4cdb5cd/frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js#L148)
- [x] Be able to get out of editing schema using keyboard shortcuts
- [x] hitting 'enter' on a list of worksheets on a worksheets does nothing #2939

### PR DESCRIPTION
**Pull request by yuqijin**
### Reasons for making this change

Fix the keyboard shortcut bug which is caused by the invalid usage of string refs. Now the following shortcut can all work normally:

- `enter`: open a worksheet
- `shift+enter`: open a worksheet in a new window
- `i`: paste the worksheet uuid to terminal

### Related issues

fixes #2939 
fixes part of #2951 

### Screenshots
- `enter`: open a worksheet

enter

- `shift+enter`: open a worksheet in a new window

shift

- `i`: paste the worksheet uuid to terminal
i

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: frontend/src/components/worksheets/items/WorksheetItem.js [hunk 1]
```diff
@@ -9,14 +9,19 @@ class WorksheetItem extends React.Component {
...
 
     capture_keys() {
         // Open worksheet in same tab
         Mousetrap.bind(
             ['enter'],
-            function(e) {
-                this.props.openWorksheet(this.refs['row' + this.props.subFocusIndex].props.uuid);
+            function() {
+                this.props.openWorksheet(
+                    this.rowRefs[
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/WorksheetItem.js]: We repeat this line four times -- can we refactor it into a function, just for readability?

Comment (yuqijin) [reply frontend/src/components/worksheets/items/WorksheetItem.js]: Which line do you refer to?

Comment (epicfaace) [reply frontend/src/components/worksheets/items/WorksheetItem.js]:
```
const uuid = this.rowRefs[
                    `codalab-worksheet-item-${this.props.focusIndex}-subitem-${this.props.subFocusIndex}`
                ].current.props.uuid;
```

Comment (yuqijin) [reply frontend/src/components/worksheets/items/WorksheetItem.js]: Thanks for the advice. I have refactored the repeated lines

**Review by epicfaace — APPROVED** [review]

**Pull request merged**